app/concepts/reservations/use_cases/create_online.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Reservations
+  module UseCases
+    class CreateOnline
+      ReservationInvalidError = Class.new(StandardError)
+
+      attr_reader :repository, :params
+
+      def initialize(params:, repository: Reservations::Repository.new)
+        @repository = repository
+        @params = params
+      end
+
+      def call
+        Reservation.transaction do
+          repository.create(reservation_params).tap do |reservation|
+            raise ReservationInvalidError, "Couldn't create reservation" unless reservation.persisted?
```

Comment:
On previews PR @rafalrothenberger told me to use `create!` or `persisted?` instead of `valid?`. I wanted to use `create!` method but I don't know how to raise a custom Error using this method. Could you help me, please?